### PR TITLE
chore(bootnode): improve bootnode settings

### PR DESCRIPTION
**Pull request by MatusKysel**

### CODE REVIEW

**Review by kstdl — APPROVED** [review]

**Review by rattrap — APPROVED** [review]
LGTM

File: programs/taraxa-bootnode/main.cpp [hunk 1]
```diff
@@ -152,7 +152,7 @@ int main(int argc, char** argv) {
 
   auto net_conf = public_ip.empty() ? dev::p2p::NetworkConfig(listen_ip, listen_port, false)
                                     : dev::p2p::NetworkConfig(public_ip, listen_ip, listen_port, false);
-  net_conf.allowLocalDiscovery = allowLocalDiscovery;
+  net_conf.allowLocalDiscovery = !denyLocalDiscovery;
```
Comment (JakubFornadel) [inline programs/taraxa-bootnode/main.cpp:155]: out of curiosity: why did you change it from allow to deny and program arg is deny but in config is allow ? :D

Comment (MatusKysel) [reply programs/taraxa-bootnode/main.cpp:155]: netconfig is also part of taraxa node, I would need to change it everywhere in code, but we need this reverse logic just for bootnodes

Comment (JakubFornadel) [reply programs/taraxa-bootnode/main.cpp:155]: yeah I am just curious what was the reasoning behind this change. Isnt having "allow" set to false same thing as "deny" to true ?

**Review by JakubFornadel — APPROVED** [review]

File: programs/taraxa-bootnode/main.cpp (hunk 1 above)
Comment (MatusKysel) [reply programs/taraxa-bootnode/main.cpp:155]: it is just because of boost program options 
```
addNetworkingOption("deny-local-discovery", po::bool_switch(&denyLocalDiscovery),
```
if you use this flag you can set it only to true

Comment (MatusKysel) [reply programs/taraxa-bootnode/main.cpp:155]: and we want to enable by default, so we need flag to disable it

Comment (JakubFornadel) [reply programs/taraxa-bootnode/main.cpp:155]: aaah ok now I get it
